akka-multi-node-testkit/src/main/scala/akka/remote/testkit/MultiNodeSpec.scala

```diff
@@ -255,9 +255,16 @@ abstract class MultiNodeSpec(val myself: RoleName, _system: ActorSystem, _roles:
 
   import MultiNodeSpec._
 
+  /**
+   * Constructor for using arbitrary logic to create the actor system used in
+   * the multi node spec (the `Config` passed to the creator must be used in
+   * the created actor system for the multi node tests to work)
+   */
+  def this(config: MultiNodeConfig, actorSystemCreator: Config ⇒ ActorSystem) =
+    this(config.myself, actorSystemCreator(ConfigFactory.load(config.config)), config.roles, config.deployments)
+
   def this(config: MultiNodeConfig) =
-    this(config.myself, ActorSystem(MultiNodeSpec.getCallerName(classOf[MultiNodeSpec]), ConfigFactory.load(config.config)),
-      config.roles, config.deployments)
+    this(config, config ⇒ ActorSystem(MultiNodeSpec.getCallerName(classOf[MultiNodeSpec]), config))
```

Comment:
Yep! However it is broken for all subclasses of `MultiNodeRemotingSpec` (by me as well, but a while ago). I created #22186 to fix that